theano/tensor/subtensor.py

```diff
@@ -967,35 +967,43 @@ def process(self, r, pstate):
         elif isinstance(r.owner.op, Subtensor):
             idxs = r.owner.op.idx_list
             inputs = list(r.owner.inputs)
-            input = inputs.pop()
+            input = inputs.pop(0)
             sidxs = []
-            inbrack_pstate = pstate.clone(precedence=-1000)
-            for entry in idxs:
-                if isinstance(entry, integer_types):
-                    sidxs.append(str(entry))
-                elif isinstance(entry, scal.Scalar):
-                    sidxs.append(inbrack_pstate.pprinter.process(inputs.pop()))
-                elif isinstance(entry, slice):
-                    if entry.start is None or entry.start == 0:
-                        msg1 = ""
-                    else:
-                        msg1 = entry.start
-
-                    if entry.stop is None or entry.stop == sys.maxsize:
-                        msg2 = ""
-                    else:
-                        msg2 = entry.stop
-
-                    if entry.step is None:
-                        msg3 = ""
-                    else:
-                        msg3 = ":%s" % entry.step
+            old_precedence = getattr(pstate, 'precedence', None)
+            try:
+                pstate.precedence = -1000
+
+                for entry in idxs:
+                    if isinstance(entry, integer_types):
+                        sidxs.append(str(entry))
+                    elif isinstance(entry, scal.Scalar):
+                        sidxs.append(pstate.pprinter.process(inputs.pop()))
+                    elif isinstance(entry, slice):
+                        if entry.start is None or entry.start == 0:
+                            msg1 = ""
+                        else:
+                            msg1 = entry.start
+
+                        if entry.stop is None or entry.stop == sys.maxsize:
+                            msg2 = ""
+                        else:
+                            msg2 = entry.stop
+
+                        if entry.step is None:
+                            msg3 = ""
+                        else:
+                            msg3 = ":%s" % entry.step
+
+                        sidxs.append("%s:%s%s" % (msg1, msg2, msg3))
+            finally:
+                pstate.precedence = old_precedence
```

Comment:
If there isn't one, it will set a new one that will be None. In the current logic, it should be equivalent. This is already what is done inside the file theano/printing.py.